/*import mongoose from "mongoose";
import supertest from "supertest";
import { expect } from "chai";
import { generar } from "./generador/usuarios.js";
import app from "../src/server.js";*/
const mongoose = require("mongoose")
const supertest = require("supertest")
const  {expect}  = require("chai")
const generar = require("./generador/productos")
const app = require( "../server.js")
require('dotenv').config() 
let request;
let server;

describe("test api rest full", () => {
  before( function () {
     connectDb();
    server =  startServer();
    request = supertest(
      `http://localhost:${server.address().port}/productos`
      
    );
  });

  /*after(function () {
    mongoose.disconnect();
    server.close();
  });*/

  describe("GET", () => {
    it("debería retornar un status 200", async () => {
      const response =  request.get("/");
      expect(response.status).to.eql(200);
    });
  });

  describe("POST", () => {
    it("debería incorporar un producto", async () => {
      const producto = generar();
      console.log("Generar",producto)
      const response =  request.post("/").send(producto);
      expect(response.status).to.eql(200);

      /*const prod = response.body;
      expect(prod).to.include.keys("title", "title");
      expect(prod.title).to.eql(producto.title);
      expect(prod.price).to.eql(producto.email);*/
    });
  });
});

function connectDb() {
  try {
    //await mongoose.connect("mongodb://localhost/mibase", {
       mongoose.connect(this.URL = process.env.MONGO_URL_ATLAS, {
      useNewUrlParser: true,
      useUnifiedTopology: true,
    });

    console.log("Base de datos conectada!");
  } catch (error) {
    throw new Error(`Error de conexión en la base de datos: ${err}`);
  }
}

async function startServer() {
  return new Promise((resolve, reject) => {
    const PORT = 0;
    const server = app.listen(PORT, () => {
      console.log(
        `Servidor express escuchando en el puerto ${server.address().port}`
      );
      resolve(server);
    });
    server.on("error", (error) => {
      console.log(`Error en Servidor: ${error}`);
      reject(error);
    });
  });
}
